smartcontract/contracts/RadaNFTToken.sol

```diff
@@ -33,29 +35,18 @@ contract RadaNFTToken is ERC721URIStorage, Ownable {
     function isWhitelister(address user) external view returns (bool) {
         return whitelist[user];
     }
-
-    function mint(string memory tokenURI) external onlyWhitelister {
+   
+    function mint(address _creator, string memory _tokenURI, uint _amount) external onlyWhitelister(_creator) {
+        
+        _mint(_creator, _tokenIds.current(), _amount, "");
         _tokenIds.increment();
+        tokenIdToURI[_tokenIds.current()] = _tokenURI;
 
-        uint256 newItemId = _tokenIds.current();
-        _safeMint(_msgSender(), newItemId);
-        _setTokenURI(newItemId, tokenURI);
     }
-
-    function batchMint(string memory tokenURI, uint256 amount)
-        external
-        onlyWhitelister
-    {
-        for (uint256 i = 0; i <= amount; i++) {
-            _tokenIds.increment();
-
-            uint256 newItemId = _tokenIds.current();
-            _safeMint(_msgSender(), newItemId);
-            _setTokenURI(newItemId, tokenURI);
-        }
-    }
-
-    function burn(uint256 tokenId) external onlyWhitelister {
-        _burn(tokenId);
+    function burn(address _creator, uint _tokenId, uint _amount ) external onlyWhitelister(_creator) {
+        _burn(_creator, _tokenId, _amount);
     }
+    function transfer(address _owner, address _receiver, uint _tokenId) external onlyWhitelister(_owner) {
+        _safeTransferFrom(_owner, _receiver, _tokenId, 1, "0x0");
```

Comment:
why 0x0 here?

Reply:
https://ethereum.stackexchange.com/questions/73125/what-is-the-bytes-data-param-in-safetransferfrom

You could check this. Anyway, this param is not important in this case.

Reply:
Just a message